packages/run-protocol/src/runStake/attestation.js

```diff
@@ -0,0 +1,254 @@
+ * @param {ZCF} zcf
+ * @param {Brand<'nat'>} stakeBrand brand of the staked assets
+ * @param {ERef<StakingAuthority>} lienBridge bridge to account state
+ */
+const makeAttestationKit = async (zcf, stakeBrand, lienBridge) => {
+  const { add, subtract, makeEmpty, isGTE, coerce } = AmountMath;
+  const empty = makeEmpty(stakeBrand);
+  const zcfMint = await zcf.makeZCFMint(KW.Attestation, AssetKind.COPY_BAG);
+  const { issuer, brand: attBrand } = zcfMint.getIssuerRecord();
+
+  /**
+   * @param {string} address
+   * @param {Amount<'nat'>} lienedAmount
+   * @returns {Amount<'copyBag'>}
+   */
+  const wrapLienedAmount = (address, lienedAmount) =>
+    AmountMath.make(attBrand, makeCopyBag([[address, lienedAmount.value]]));
+
+  /** @param {Amount<'copyBag'>} attestationAmount */
+  const unwrapLienedAmount = attestationAmount => {
+    assert(
+      attestationAmount.brand === attBrand,
+      X`The escrowed attestation ${attestationAmount} was not of the attestation brand ${attBrand}`,
+    );
+    fit(attestationAmount.value.payload, harden([[M.string(), M.bigint()]]));
+    /** @type {[string, bigint][]} */
+    const [[address, valueReturned]] = attestationAmount.value.payload;
+    const lienedAmount = AmountMath.make(stakeBrand, valueReturned);
+    return { address, lienedAmount };
+  };
+
+  /** @type {(m: Amount, s: Amount) => Amount} */
+  const subtractOrZero = (m, s) => (isGTE(m, s) ? subtract(m, s) : empty);
+  // check that x <= hi - lo
+  const checkDelta = (x, label, hi, lo) => {
+    const bound = subtractOrZero(hi, lo);
+    assert(
+      isGTE(bound, coerce(stakeBrand, x)),
+      // TODO: X`...${raw(label)}...`
+      X`Only ${bound} was ${label}, but an attestation was attempted for ${x}`,
+    );
+  };
+  const assertAccountStateOK = async (address, toLien) => {
+    const s = await E(lienBridge).getAccountState(address, stakeBrand);
+    const unlocked = subtractOrZero(s.total, s.locked);
+    checkDelta(toLien, 'unliened', s.total, s.liened);
+    checkDelta(toLien, 'bonded and unliened', s.bonded, s.liened);
+    checkDelta(toLien, 'unlocked and unliened', unlocked, s.liened);
+  };
+
+  /** @type {Store<Address, Amount<'nat'>>} */
+  const amountByAddress = makeStore('amount');
+
+  /**
+   * @param { string } address
+   * @param { Amount<'nat'> } lienDelta
+   */
+  const mintAttestation = async (address, lienDelta) => {
+    assert.typeof(address, 'string');
+    // This account state check is primarily to provide useful diagnostics.
+    // Since things like the bonded amount can change out from under us,
+    // we shouldn't rely on it completely.
+    // TODO: consider performance implications vs. diagnostics.
+    await assertAccountStateOK(address, lienDelta);
+
+    const current = getOrElse(amountByAddress, address, () => empty);
+    const target = add(current, lienDelta);
+
+    // Since our mint is not subject to normal supply/demand,
+    // minting here and failing in setLiened below does no harm.
```

Comment:
I think by design nothing would break. I'm just uneasy about the amount of moving parts in a call to `mintZCFMintPayment` so I was avoiding moving them after the commit point.

Reply:
the `mintZCFMintPayment` call includes remote operations, which could fail, so we agreed this must go before the commit point.